plugin/flags_sources/compilation_db.py

```diff
@@ -127,13 +128,17 @@ def _parse_database(self, database_file):
         parsed_db = {}
         unique_list_of_flags = UniqueList()
         for entry in data:
-            file_path = File.canonical_path(entry['file'],
-                                            database_file.folder())
             argument_list = []
 
-            base_path = database_file.folder()
+            file_path = entry['file']
```

Comment:
No need for this, `file_path` is anyway set later

Reply:
No, its not. Its not set if the file_path is absolut.